core/include/detray/builders/grid_factory.hpp

```diff
@@ -96,13 +103,23 @@ class grid_factory {
         constexpr auto e_phi_axis = static_cast<dindex>(axes_t::label1);
 
         auto b_values = grid_bounds.values();
+        // Overwrite the mask values if axis spans are provided
+        if (axis_spans[0].size() > 1) {
+            b_values[boundary::e_min_r] = axis_spans[0][0];
```

Comment:
It might make sense to use ```at``` here, as it provides bounds checks and makes finding problems during the rest of the IO chain easier:
```suggestion
            b_values[boundary::e_min_r] = axis_spans[0].at(0);
```

Reply:
And for all other overloads, too